airflow/providers/google/cloud/operators/dataprocgdc.py

```diff
@@ -0,0 +1,300 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+"""This module contains Dataproc GDC operators using KRM APIs."""
+
+from __future__ import annotations
+
+import datetime
+from functools import cached_property
+from typing import Sequence, TYPE_CHECKING
+
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.providers.cncf.kubernetes.hooks.kubernetes import _load_body_to_dict
+from airflow.providers.cncf.kubernetes.hooks.kubernetes import KubernetesHook
+from kubernetes.client import CoreV1Api, ApiException
+from kubernetes.watch import Watch
+
+
+if TYPE_CHECKING:
+    from kubernetes.client.models import CoreV1EventList
+    from airflow.utils.context import Context
+
+
+class DataprocGDCSubmitSparkJobKrmOperator(BaseOperator):
```

Comment:
Can't we extend the functionality of existing operator `DataprocSubmitSparkJobOperator` ?

Reply:
There are two ways we need to go for implementing Airflow operator to trigger workloads on GDC cluster.
1 is via KRM API and another is via Google cloud API(where we could use the DataprocSubmitSparkJobOperator, however need to check if that's possible as well).

This PR targets the KRM APIs exposed in a GDC cluster. So this cannot leverage the existing operator.